code/obj/item/misc_weapons.dm

```diff
@@ -1189,18 +1189,13 @@
 
 	attack_hand(mob/living/carbon/human/user as mob)
 		if(user.r_hand == src || user.l_hand == src || user.belt == src)
-			if(src.sword_inside) //Checks if a katana is inside
-				sword_inside.clean_forensic()
-				boutput(user, "You draw [sword_inside] from your sheath.")
-				playsound(get_turf(user), pick("sound/effects/sword_unsheath1.ogg","sound/effects/sword_unsheath2.ogg"), 70, 0, 0)
-				icon_state = sheath_state
-				item_state = ih_sheath_state
-				user.put_in_hand_or_drop(sword_inside)
-				sword_inside = null //No more sword inside.
-				user.update_clothing()
-			else
-				return ..()//Katana doesn't exist, and takes the sheath off your belt or switches hands.
+			draw_sword(user)
+		else
+			return ..()
 
+	attack_self(mob/living/carbon/human/user as mob)
+		if(user.r_hand == src || user.l_hand == src || user.belt == src)
```

Comment:
This is really trivial, but the sheath should never be in the user's belt slot when it is calling its attack_self proc, so it's pointless to have the check here. Of course, it will never be executed due to the short circuiting "||" operators on r_hand and l_hand. But still, it's unnecessary.